### PR TITLE
Adds JibAdapter and JibMavenPluginAdapter.

### PR DESCRIPTION
**Pull request by coollog**
Part of #13

### CODE REVIEW

File: skaffold-maven-plugin/src/test/java/com/google/cloud/tools/skaffold/jib/JibMavenPluginAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link JibMavenPluginAdapter}. */
+@RunWith(MockitoJUnitRunner.class)
+public class JibMavenPluginAdapterTest {
+
+  private static Xpp3Dom buildXpp3Dom(String xml) throws IOException, XmlPullParserException {
+    return Xpp3DomBuilder.build(
+        new InputStreamReader(
```
Comment (briandealwis) [inline skaffold-maven-plugin/src/test/java/com/google/cloud/tools/skaffold/jib/JibMavenPluginAdapterTest.java:47]: StringReader?

**Review by briandealwis — COMMENTED** [review]
I'm thinking throwing exceptions might make more sense here.

It seems odd for a project with no  Jib `<plugin>` section to get a `ResolvedJib` instance when it's not actually resolved.

What should a plugin do if `isVersionSupported()` is false?  The safest thing is to abort, right?  So may as well turn this situation into an exception, don't you think?

Maybe:
```
String imageReference = null;
try {
   ResolvedJib adapter = JibMavenPluginAdapter.adapt(mavenProject);
  if(imagerReference == null) {
    imageReference = adapter.getImageReference();
  }
} catch(UnsupportVersionException ex) {
  System.err.println(ex);
  throw ex;
} catch(JibNotReferencedException ex) {
   // whattodo
}
```

**Comment by coollog** [comment]
I was thinking of handling `isVersionSupported==false` with giving a warning to users about the version of Jib being unsupported. For example, if `jib-maven-plugin` was version `0.10.0`, `skaffold-maven-plugin` that only supports up to `0.9.x` would warn that the `0.10.0` version was not supported, but still try to use the adapter for the `0.9.x` versions to find `to.image`. This will allow for new releases of `jib-maven-plugin` to not immediately break `skaffold-maven-plugin` that has not been updated yet.

For Jib not found, I'm trying to keep this codebase lighter on exceptions, since those added a lot of catches and throws everywhere in the Jib codebase - and only have exceptions for cases that are truly exceptions and not something that is expected (like not having `jib-maven-plugin`). Giving back a `ResolvedJib` will allow the caller to decide how to handle the resolution result without using messy try-catch blocks. Does this sound reasonable?

**Comment by coollog** [comment]
Example use:

```
ResolvedJib resolution = JibMavenPluginAdapter.resolve(mavenProject);
if (!resolution.hasJib()) {
  return tryOtherWaysToGetTargetImageReference();
}
if (!resolution.isSupportedVersion()) {
  warn("Jib version is not supported");
}
if (!resolution.getImageReference.isPresent()) {
  warn("Jib present but target image reference not defined");
  return tryOtherWaysToGetTargetImageReference();
}
return resolution.getImageReference().get();
```

**Comment by briandealwis** [comment]
That makes sense.  I think my confusion comes from the use  of `Resolved` in `ResolvedJib` which suggests that the resolution was successful.  Would `ResolvedJib` &rarr; `JibParameters` work?

**Pull request closed without merging**